lib/capybara/webkit/node.rb

```diff
@@ -15,16 +15,19 @@ def all_text
     end
 
     def [](name)
-      value = invoke("attribute", name)
-      if name == 'checked' || name == 'disabled' || name == 'multiple'
-        value == 'true'
-      else
-        if invoke("hasAttribute", name) == 'true'
-          value
-        else
-          nil
-        end
+      name = name.to_s
+      # Although the attribute matters, the property is consistent. Return that in
+      # preference to the attribute for links and images.
+      tn = tag_name
+      if (tn == "img" && name == "src") || (tn == "a" && name == "href")
+        # if attribute exists get the property
+        val = invoke(:attribute, name) && invoke(:property, name)
+        return val
```

Comment:
Could you use `else` instead of an eager `return`? I find it easier to parse the possible paths when there aren't escape hatches.